addons/mail/static/src/core/common/thread_model.js

```diff
@@ -983,6 +983,17 @@ export class Thread extends Record {
         }
     }
 
+    replaceMessages(tmpMsg, message) {
+        // The message from other personas (not self) should not replace the tmpMsg
+        if (tmpMsg && tmpMsg.in(this.messages) && message.author.eq(this.store.self)) {
```

Comment:
Do we actually need to check for author? It looks like `temporary_id` should only be set/only be sent for the current user naturally (other persons can't be aware of a temporary message).